### PR TITLE
Add validation to check if time series data streams have dimensions

### LINKED ISSUE #458: [Change proposal] Validate data stream with index_mode:time_series contains dimension field
## Description 

We now honor `index_mode:time_series` in Fleet in Kibana, but it's possible to declare a datastream with index_mode:time_series and no dimension fields. 

It will be great if this could be validated during package build.

Related error 
Screen Shot 2023-01-09 at 11 25 15 AM

cc @ruflin

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

Add a validation to check that data streams with time series enabled have at least a dimension.

## Why is it important?

These data streams are invalid, and rejected by Elasticsearch, detect the issue earlier.

See #458.

## Checklist

- [x] I have added test packages to [`test/packages`](https://github.com/elastic/package-spec/tree/main/test/packages) that prove my change is effective.
- [x] I have added an entry in [`spec/changelog.yml`](https://github.com/elastic/package-spec/blob/main/spec/changelog.yml).

## Related issues

- Fixes #458

### CODE REVIEW

File: code/go/internal/validator/spec.go [hunk 1]
```diff
@@ -163,6 +163,7 @@ func (s Spec) rules(pkgType string, rootSpec spectypes.ItemSpec) validationRules
 		{fn: semantic.ValidateKibanaNoDanglingObjectIDs, since: semver.MustParse("3.0.0")},
 		{fn: semantic.ValidateKibanaFilterPresent, since: semver.MustParse("3.0.0")},
 		{fn: semantic.ValidateKibanaNoLegacyVisualizations, types: []string{"integration"}, since: semver.MustParse("3.0.0")},
+		{fn: semantic.ValidateDimensionsPresent, types: []string{"integration"}, since: semver.MustParse("3.0.1")},
```
Comment (mrodm) [inline code/go/internal/validator/spec.go:166]: :+1:

**Review by mrodm — APPROVED** [review]
LGTM
Just a minor comment

File: code/go/internal/validator/semantic/validate_dimensions_present.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+	}
+	for _, dataStream := range dataStreams {
+		tsEnabled, err := isTimeSeriesModeEnabled(fsys, dataStream)
+		if err != nil {
+			return specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+		}
+		_, hasDimensions := dimensionPresent[dataStream]
+		fmt.Println(dataStream, tsEnabled, hasDimensions)
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply code/go/internal/validator/semantic/validate_dimensions_present.go:40]: Indeed it is, thanks!

**Comment by jsoriano** [comment]
test integrations

**Comment by jsoriano** [comment]
Failures in https://github.com/elastic/integrations/pull/8357 probably related to environment and not to this change. Tested validation in all packages locally and they pass, I will merge this change.

**Pull request merged**